File: TypeScript/test/jest/mailServiceMock.ts
import { IMailService } from "@/IMailService";

export default class mailServiceMock implements IMailService {
  private expectedErrorType!: number;
  private actualErrorType!: number;

  sendLoginFailureMail(userName: string): void {}

  sendSystemFailureMail(errorType: number): void {
    this.expectedErrorType = errorType;
  }

  expectFailureMail(errorType: number): void {
    this.expectedErrorType = errorType;
  }

  public verify(): void {
    expect(this.actualErrorType).toBe(this.expectedErrorType);
  }
}

File: CSharpCore/CSharpCoreTest/MailServiceMock.cs
using System;
using System.Collections.Generic;
using System.Text;
using FluentAssertions;

namespace CSharpCore.Test
{
    public partial class MailServiceMock : IMailService
    {
        private int expectedErrorType;
        private int actualErrorType;

        public void SendLoginFailureMail(String userName)
        {

        }


        public void SendSystemFailureMail(int errorType)
        {
            actualErrorType = errorType;
        }

        public void ExpectFailureMail(int errorType)
        {
            this.expectedErrorType = errorType;
        }

        public void Verify()
        {
            actualErrorType.Should().Be(expectedErrorType);
        }
    }
}
